source/s3_auto_ranged_get.c

```diff
@@ -277,37 +275,28 @@ static bool s_s3_auto_ranged_get_update(
 
     aws_s3_meta_request_unlock_synced_data(meta_request);
 
-    if (!work_remaining) {
+    if (work_remaining) {
+        if (request != NULL) {
+            AWS_ASSERT(out_request != NULL);
```

Comment:
Unfortunately, the interface is a bit awkward right now.  Right now, we have a case where we update the meta request, but can't send requests.  In that case, it's valid to pass in NULL, signaling to the function, "update, but don't pass anything back to me, because I couldn't do anything with it."  So this assert is essentially, "I didn't create a request when I wasn't supposed to."

Reply:
Ok, maybe just comment to that effect, since the next person in here is going to be confused as hell like I was :P